flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/HBaseConnectorITCase.java

```diff
@@ -20,281 +18,464 @@
+		tEnv.registerTableSource("hTable", hbaseTable);
 
-		Table result = tableEnv.sqlQuery(
-			"SELECT * FROM hTable AS h"
-		);
-		DataSet<Row> resultSet = tableEnv.toDataSet(result, Row.class);
-		List<Row> results = resultSet.collect();
+		Table table = tEnv.sqlQuery("SELECT * FROM hTable AS h");
 
+		List<Row> results = collectBatchResult(table);
 		String expected =
 			"Hello-1,100,1.01,false,Welt-1,10\n" +
-			"Hello-2,200,2.02,true,Welt-2,20\n" +
-			"Hello-3,300,3.03,false,Welt-3,30\n" +
-			"null,400,4.04,true,Welt-4,40\n" +
-			"Hello-5,500,5.05,false,Welt-5,50\n" +
-			"Hello-6,600,6.06,true,Welt-6,60\n" +
-			"Hello-7,700,7.07,false,Welt-7,70\n" +
-			"null,800,8.08,true,Welt-8,80\n";
+				"Hello-2,200,2.02,true,Welt-2,20\n" +
+				"Hello-3,300,3.03,false,Welt-3,30\n" +
+				"null,400,4.04,true,Welt-4,40\n" +
+				"Hello-5,500,5.05,false,Welt-5,50\n" +
+				"Hello-6,600,6.06,true,Welt-6,60\n" +
+				"Hello-7,700,7.07,false,Welt-7,70\n" +
+				"null,800,8.08,true,Welt-8,80\n";
 
 		TestBaseUtils.compareResultAsText(results, expected);
 	}
 
 	@Test
 	public void testTableSourceReadAsByteArray() throws Exception {
-
-		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
-		env.setParallelism(4);
-		BatchTableEnvironment tableEnv = BatchTableEnvironment.create(env, new TableConfig());
+		TableEnvironment tEnv = createBatchTableEnv();
 		// fetch row2 from the table till the end
-		HBaseTableSource hbaseTable = new HBaseTableSource(getConf(), TEST_TABLE);
+		HBaseTableSource hbaseTable = new HBaseTableSource(getConf(), TEST_TABLE_1);
 		hbaseTable.addColumn(FAMILY2, F2COL1, byte[].class);
 		hbaseTable.addColumn(FAMILY2, F2COL2, byte[].class);
+		tEnv.registerTableSource("hTable", hbaseTable);
+		tEnv.registerFunction("toUTF8", new ToUTF8());
+		tEnv.registerFunction("toLong", new ToLong());
 
-		tableEnv.registerTableSource("hTable", hbaseTable);
-		tableEnv.registerFunction("toUTF8", new ToUTF8());
-		tableEnv.registerFunction("toLong", new ToLong());
-
-		Table result = tableEnv.sqlQuery(
+		Table table = tEnv.sqlQuery(
 			"SELECT " +
 				"  toUTF8(h.family2.col1), " +
 				"  toLong(h.family2.col2) " +
 				"FROM hTable AS h"
 		);
-		DataSet<Row> resultSet = tableEnv.toDataSet(result, Row.class);
-		List<Row> results = resultSet.collect();
 
+		List<Row> results = collectBatchResult(table);
 		String expected =
 			"Hello-1,100\n" +
-			"Hello-2,200\n" +
-			"Hello-3,300\n" +
-			"null,400\n" +
-			"Hello-5,500\n" +
-			"Hello-6,600\n" +
-			"Hello-7,700\n" +
-			"null,800\n";
+				"Hello-2,200\n" +
+				"Hello-3,300\n" +
+				"null,400\n" +
+				"Hello-5,500\n" +
+				"Hello-6,600\n" +
+				"Hello-7,700\n" +
+				"null,800\n";
+
+		TestBaseUtils.compareResultAsText(results, expected);
+	}
+
+	@Test
+	public void testTableInputFormat() throws Exception {
+		if (BLINK_PLANNER.equals(planner)) {
+			// this case is for testing TableInputFormat which is not works for flink-table
```

Comment:
This comment is a little bit ambiguous, from the code logic it seems we should say "this case is for testing TableInputFormat which only applies to flink-table planner" or so?

Reply:
Not exactly. I mean `TableInputFormat` is a runtime implementation, not for flink table, not relative to what the planner is. You can see the test is only verified on `DataSet`, not on `TableEnvironment`.

Reply:
After some offline discussion, the case is irrelative to planner actually, but in case that later change somehow making the test case relative to planner implementation (although low possibility), let's just don't skip the test for any planner.